config/archive_test.go

```diff
@@ -0,0 +1,120 @@
+package config
+
+import (
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/contiv/volplugin/watch"
+	. "gopkg.in/check.v1"
+)
+
+const (
+	testPolicyName = "foobar"
+)
+
+func revisionCount(s *configSuite) int {
+	revisions, _ := s.tlc.ListPolicyRevisions(testPolicyName)
+	return len(revisions)
+}
+
+func createTestPolicy(s *configSuite, c *C) {
+	err := s.tlc.CreatePolicyRevision(testPolicyName, "{}")
+	c.Assert(err, IsNil)
+}
+
+func (s *configSuite) TestCreatePolicyRevision(c *C) {
+	c.Assert(revisionCount(s), Equals, 0)
+	createTestPolicy(s, c)
+	c.Assert(revisionCount(s), Equals, 1)
+}
+
+func (s *configSuite) TestListPolicyRevisions(c *C) {
+	c.Assert(revisionCount(s), Equals, 0)
+
+	// edge case: the second may roll over between saving the current timestamp
+	//            and the HTTP request to upload the policy being handled
+	t := int(time.Now().Unix())
```

Comment:
If we don't check the timestamp in either of the tests, then testing the `TestCreatePolicyRevision()` and `TestListPolicyRevisions()` functions will have identical tests (basically just checking that the # of items being returned has changed by the expected amount).

@yuva29  I like the the idea to use `GetPolicyRevision()`.

I think `TestCreatePolicyRevision()` should test that the created policy entry is under the expected key name (the UNIX timestamp at creation time), and the `ListPolicyRevisions()` should test that all the keys in the response refer to real revisions.  Only one test (i.e., creation) really needs to test that the etcd entry has the correct name.

What do you guys think?